rbac/api/cross_access/serializer.py

```diff
@@ -95,3 +95,18 @@ def create(self, validated_data):
         for role in roles:
             request.roles.add(role)
         return request
+
+    def update(self, instance, validated_data):
+        """Override the update method to associate the roles to cross account request after it is updated."""
+        if "roles" in validated_data:
+            role_data = validated_data.pop("roles")
+            display_names = [role["display_name"] for role in role_data]
+            roles = Role.objects.filter(display_name__in=display_names)
+            instance.roles.clear()
```

Comment:
Were we still thinking of wrapping this all in a transaction? Maybe the `update` method already does that?

Reply:
Did a bit of digging, it's hard to determine if a serializer update() is automatically wrapped in a transaction. I see here [1] that some query attached updates do, but that doesn't seem to extend to an update of this style. I think I'll go ahead and wrap it and we'll see how it behaves.

[1]: https://docs.djangoproject.com/en/3.1/topics/db/transactions/

Reply:
A second bit of digging further down the page shows that we could enable ATOMIC_REQUESTS and have all of our HTTP requests take place in a transaction, but as that's a larger change, I'm still thinking I'll just wrap this for now.